fili-core/src/main/java/com/yahoo/bard/webservice/table/util/ImmutableWrapperMap.java

```diff
@@ -0,0 +1,90 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A map wrapping implementation to support fully immutable availabilities.
+ *
+ * @param <K> The map key type
+ * @param <V> The map value type
+ */
+public class ImmutableWrapperMap<K, V> implements Map<K, V> {
+
+    Map<K, V> map;
```

Comment:
Just curious, if we implement `Map`, does it make sense to have this? Looks like `HashMap` doesn't do it this way http://grepcode.com/file_/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/util/HashMap.java/?v=source